admin/admin-functions.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * The admin-specific functionality of the plugin.
+ *
+ * @link       https://nataliesmith.dev
+ * @since      1.0.0
+ * @package    Poetwp
+ * @subpackage Poetwp/admin
+ */
+
+/**
+ * Add custom meta boxes to the poem post type
+ */
+function poetwp_add_meta_boxes() {
+	add_meta_box( 'poem_details', 'Poem Details', 'poetwp_poem_details_callback', 'poem', 'normal', 'default' );
+}
+// Hook the meta box function to the 'add_meta_boxes' action.
```

Comment:
Do you mean just the hook specific // comments, or any of the "internal" // comments, like here?

```
function poetwp_hide_default_title( $title, $id = null ) {
	// Only hide title for 'poem' post type on single post pages and in the main loop.
	if ( is_singular( 'poem' ) && in_the_loop() && get_post_type( $id ) === 'poem' ) {
		return '';
	}
	return $title;
}
```

Most of these comments are for me, they help me understand the code better while I'm building the plugin. The plan was to remove them all once I'm done.

Reply:
In this case, I only meant the hook comments.

But your example can also be talked about.

The trick here is to [decompose the conditional](https://refactoring.guru/decompose-conditional), i.e. if the condition is so complex that it needs a comment to understand it, then move that logic into its own helpfully named method that can have its own DocBlock:

```php
/**
 * Determine if post title should be hidden.
 *
 * Only hide title for 'poem' post type on single post pages and in the main loop.
 */
function poetwp_should_hide_title( $id ): bool {
	return is_singular( 'poem' ) && in_the_loop() && get_post_type( $id ) === 'poem' );
}
```

Then the logic in your example becomes:

```php
function poetwp_hide_default_title( $title, $id = null ) {
	return poetwp_should_hide_title() ? '' : $title;
}
```

What we've done is to split out the logic to:

 - decide if a post title should be hidden
 - decide on the output if a post title should be hidden

Two responsibilities, two functions - a la Single Responsibility Principle.

If you later determine that the title should be hidden under other conditions, you update the `poetwp_should_hide_title()` with additional logic as needed.

If you write a `//` comment, consider it a code smell that might be an opportunity to extract the logic into a named function/method.

Reply:
Ok thanks. If I add the hook above the callback function, as per your other comment, this will be resolved anyway. And I will look into my internal comments as well